Top/argdecode.c

```diff
@@ -1557,8 +1557,11 @@ PUBLIC int argdecode(CSOUND *csound, int argc, const char **argv_)
               break;
             }
 #endif
-            if (!decode_long(csound, s, argc, argv))
-              csound->LongJmp(csound, 1);
+           if (!decode_long(csound, s, argc, argv))
+              // shouldn't need to long jump here -
+              // - it's crashing if csoundSetOption fails
+              //csound->LongJmp(csound, 1);
+             csound->Message(csound,"\n ...ignoring \n");
```

Comment:
Think we need to check the behavior here. If a user supplies an invalid option, this function should return an error, yes? That way the commandline app can exit gracefully and API apps can handle that situation. (If I'm reading this code correctly...)

Reply:
this was a bug fix. It was crashing csound when the API function was called on non-existent option strings. We can look at this later. Sometimes exiting is not the right thing. It can still carry on, with a warning. Long jumps will cause trouble.

Reply:
I have no problem with getting rid of longjmps as I don't care for it, but this fix for setOption breaks the commandline IMO. I just tried running csound with an invalid long option and it not only kept running, it looks like it tried to parse the flag twice:

```
stevenyi@stevens-MacBook-Pro build % ./csound ~/work/csound/live-code/livecode.csd --asdsdfsdf 
unknown long option: '--asdsdfsdf'
 ...ignoring 
Reading options from $HOME/.csound7rc
UnifiedCSD:  /Users/stevenyi/work/csound/live-code/livecode.csd
unknown long option: '--asdsdfsdf'
 ...ignoring 
scoreless operation
```

I can't think of a CLI app that doesn't fail on invalid args, so this is strange, plus the double "unknown long option" is concerning. 

Took a look at the code a bit further, there's on a few uses of argdecode in the codebase. Even though argdecode returns an int, it only ever returns 1. csoundSetOption then flips the result to a 0 with a strange ternary. Additionally, there are other calls to dieu(), which in turn calls longjmp. 

It looks like at one point argdecode *did* actually return a value as main.c checks the result of argdecode and will call longjmp there (which is fine as it's the CLI app), but that actually never happens since argdecode was calling longjmp first. 

I think we should use argdecode's return value in these areas of code and have the code here return 0 for a failure. We should also replace calls to dieu() with return 0.

Reply:
this issue is really unrelated to deinit(). I saw the bug and prevented the crash. Can we file a ticket and continue? I think it needs to be considered separately.

Reply:
I think that's fine. I added https://github.com/csound/csound/issues/1878 to track this with reference to this thread here. Shouldn't take too much to sort out I think but happy enough to get this PR done first and address this in that issue.